enderio-base/src/main/java/crazypants/enderio/base/filter/items/ItemBasicItemFilter.java

```diff
@@ -92,4 +105,33 @@ public void addInformation(@Nonnull ItemStack stack, @Nullable World worldIn, @N
     }
   }
 
+  @Override
+  @Nonnull
+  public ActionResult<ItemStack> onItemRightClick(@Nonnull World worldIn, @Nonnull EntityPlayer playerIn, @Nonnull EnumHand handIn) {
+    if (playerIn.isSneaking() && ModObjectRegistry.getModObjectNN(this).openGui(worldIn, playerIn.getPosition(), playerIn)) {
```

Comment:
you may want to put the hand into openGui instead of the player position (the blockpos parameter is optional, I wouldn't put in data that is not needed just because the param is there) instead of hardcoding mainHand. Afterall, this method will be called for off-hand istemstacks, too.